### ISSUE TITLE
Math expressions in a nested list are not syntax highlighted or rendered in preview in RStudio editor for rmarkdown

### ISSUE THREAD

**Issue by zenggyu** [rstudio/rstudio#2370]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.414
    OS Version      : Ubuntu 16.04
    R Version       : 3.4.3

### Steps to reproduce the problem

Copy and paste the following text to an rmarkdown file that is opened by the RStudio editor:

```
Nested unordered list:

- list level 1 $formula 1$
    - list level 2 $formula 2$
        - list level 3 $formula 3$

Nested ordered list:

1. list level 1 $formula 1$
    2. list level 2 $formula 2$
        3. list level 3 $formula 3$
```

### Describe the problem in detail

Math expressions in list level 2 or above (i.e., $formula 2$, $formula 3$) won't be syntax highlighted, and won't be rendered in preview in the editor.

### Describe the behavior you expected

$formula 2$ and $formula 3$ should be syntax highlighted and rendered in preview in the RStudio editor

**Comment by kevinushey** [comment]
Thanks for the bug report! I've reproduced this and we'll see if we can get this fixed up.

![screen shot 2018-03-12 at 8 37 27 am](https://user-images.githubusercontent.com/1976582/37293496-b37eec6a-25d0-11e8-83a6-57e6a01c9d77.png)

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
